nixos/modules/services/misc/transfer-sh.nix

```diff
@@ -0,0 +1,434 @@
+          EnvironmentFile = lib.mkIf (cfg.environmentFile != null) cfg.environmentFile;
+        };
+
+        environment =
+          let
+            boolString = s: if s then "true" else "false";
+          in
+          {
+            LISTENER = "${cfg.address}:${builtins.toString cfg.LISTENER}";
+            PROFILE_LISTENER = builtins.toString cfg.PROFILE_LISTENER;
+            HTTP_AUTH_USER = cfg.HTTP_AUTH_USER;
+            HTTP_AUTH_PASS = cfg.HTTP_AUTH_PASS;
+            HTTP_AUTH_HTPASSWD = cfg.HTTP_AUTH_HTPASSWD;
+            HTTP_AUTH_IP_WHITELIST = cfg.HTTP_AUTH_IP_WHITELIST;
+            IP_WHITELIST = cfg.IP_WHITELIST;
+            IP_BLACKLIST = cfg.IP_BLACKLIST;
+            TEMP_PATH = cfg.TEMP_PATH;
+            WEB_PATH = cfg.WEB_PATH;
+            PROXY_PATH = cfg.PROXY_PATH;
+            PROXY_PORT = builtins.toString cfg.PROXY_PORT;
+            EMAIL_CONTACT = cfg.EMAIL_CONTACT;
+            GA_KEY = cfg.GA_KEY;
+            USERVOICE_KEY = cfg.USERVOICE_KEY;
+            HOSTS = cfg.HOSTS;
+            LOG = cfg.LOG;
+            CORS_DOMAINS = cfg.CORS_DOMAINS;
+            CLAMAV_HOST = cfg.CLAMAV_HOST;
+            PERFORM_CLAMAV_PRESCAN = boolString cfg.PERFORM_CLAMAV_PRESCAN;
+            RATE_LIMIT = cfg.RATE_LIMIT;
+            MAX_UPLOAD_SIZE = cfg.MAX_UPLOAD_SIZE;
+            PURGE_DAYS = cfg.PURGE_DAYS;
+            RANDOM_TOKEN_LENGTH = cfg.RANDOM_TOKEN_LENGTH;
+            BASEDIR = cfg.BASEDIR;
+            PURGE_INTERVAL = builtins.toString cfg.PURGE_INTERVAL;
+          } //
+
+          # Options specific to s3 backend
+          (if cfg.provider == "s3" then {
+            AWS_ACCESS_KEY = cfg.AWS_ACCESS_KEY;
+            AWS_SECRET_KEY = cfg.AWS_SECRET_KEY;
+            BUCKET = cfg.BUCKET;
+            S3_REGION = cfg.S3_REGION;
+            S3_ENDPOINT = cfg.S3_ENDPOINT;
+            S3_NO_MULTIPART = boolString cfg.S3_NO_MULTIPART;
+            S3_PATH_STYLE = boolString cfg.S3_PATH_STYLE;
+
+            # Options specific to storj backend
+          } else { }) // (if cfg.provider == "storj" then {
+            STORJ_ACCESS = cfg.STORJ_ACCESS;
+            STORJ_BUCKET = cfg.STORJ_BUCKET;
+
+            # Options specific to google drive backend
+          } else { }) // (if cfg.provider == "gdrive" then {
+            GDRIVE_CLIENT_JSON_FILEPATH = cfg.GDRIVE_CLIENT_JSON_FILEPATH;
+            GDRIVE_LOCAL_CONFIG_PATH = cfg.GDRIVE_LOCAL_CONFIG_PATH;
+            GDRIVE_CHUNK_SIZE = cfg.GDRIVE_CHUNK_SIZE;
+          } else { }) //
+
+          # TLS specific options
+          (if cfg.TLS_LISTENER != null then {
+            TLS_LISTENER = "${cfg.address}:${builtins.toString cfg.TLS_LISTENER}";
+            TLS_LISTENER_ONLY = cfg.TLS_LISTENER_ONLY;
+            TLS_CERT_FILE = cfg.TLS_CERT_FILE;
+            TLS_PRIVATE_KEY = cfg.TLS_PRIVATE_KEY;
+            FORCE_HTTPS = cfg.FORCE_HTTPS;
+          }
+          else { });
```

Comment:
This `if ... then { ... } else {}` could use `lib.optionalAttrs`.

Reply:
fixed